test/e2e/specs/fakerp/metricsbridge.go

```diff
@@ -0,0 +1,108 @@
+package fakerp
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ghodss/yaml"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/openshift-azure/pkg/util/roundtrippers"
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+// Test checks if all configured queries are valid prometheus queries, and if the metric exists
+// Existence of metric is checked by either:
+//  - query returning data
+//  - query being on a list of manually verified correct queries (that normally don't return any data)
+//  - query having ' != 0' at the end, that returns data after this condition is removed
+//  - query calculating rate( x [1m]) of a metric x, when querying for metric x returns data
+var _ = Describe("Metricsbridge E2E check configured queries [Fake][EveryPR]", func() {
+	// queries which usually return no datapoints, but have been verified as valid manually
+	manuallyVerifiedQueries := map[string]bool{
+		"kubelet_runtime_operations_errors != 0": true,
+		"kubelet_docker_operations_errors != 0":  true,
+	}
+	It("should be possible to get data for each metric in defined queries", func() {
+		By("getting the configMap")
+		mbConfigMap, err := sanity.Checker.Client.Admin.CoreV1.ConfigMaps("openshift-azure-monitoring").Get("metrics-bridge", meta_v1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mbConfigMap).ToNot(BeNil())
+
+		var mbConfig struct {
```

Comment:
much of the code from here to line 76 is duplicate, please can you library-ise it?

Reply:
changed to using the struct in pkg/entrypoint/metricsbridge instead.